### ISSUE TITLE
Strange behaviour on master system at boot

### ISSUE THREAD

**Issue by fmaida** [libretro/Genesis-Plus-GX#305]
I'm using Genesis Plus GX to emulate the Sega Master system. I have also enabled the feature of using an original bootloader rom before launching every rom.
The first time I launch a game, everything works as expected: I see the SMS boot intro and then the game on cartridge starts.

But if I go into the quick menu, choose to reset the cartridge and keep the two button pressed on the gamepad, I'm seeing this strange behavior: The audio from the boot rom continues to play, while on video I can see the game ROM

If I leave unpressed the buttons on the gamepad, the video reverts back to the boot screen.

I have made a short video to show you what happen precisely:

https://user-images.githubusercontent.com/605017/182816737-4dd0ee08-2adc-459c-8991-e1f2a26d3548.mp4

**Comment by Ryunam** [comment]
Can you check whether you have Runahead enabled (either in Single Instance or Second Instance mode)? It may be interfering with the reset function and show an incorrect image while the buttons are being pressed.

**Comment by fmaida** [comment]
Yes, I have run-ahead enabled with also the second instance enabled.
I tried to disable the run-ahead and actually the problem no longer arises.

**Comment by Ryunam** [comment]
Eh, I figured it was because of that! The entire functionality of Runahead is based on continuously reloading savestates. As such, it is no surprise that you have encountered this little quirk, as it is a type of issue that may sometimes come up when using Runahead with some of the available cores.

I think you should be able to prevent this from happening by keeping Runahead active and disabling Second Instance mode for this core. Let me know if this sorts it out.

[Piccola parentesi: immagino siamo connazionali, a giudicare dal tuo video! :) Ero tentato in effetti di replicarti direttamente in italiano, ma penso sia preferibile proseguire in inglese, a beneficio di altri utenti che possano eventualmente incappare nella stessa anomalia]

**Comment by fmaida** [comment]
Excellent, thank you very much. Grazie mille: si in effetti anch'io sono un italiano.
